.ci/setup.sh

```diff
@@ -1,6 +1,7 @@
 #!/bin/bash
 
 if [[ $OS_NAME == "macos" ]]; then
+    brew update-reset && brew update
```

Comment:
Please move this temp line to 
https://github.com/microsoft/LightGBM/blob/d517ba12f2e7862ac533908304dddbd770655d2b/.ci/test_r_package.sh#L79
with the aim to not slow down non-R jobs.

Reply:
oh I don't really think of it as temporary or as R-specific. What is currently happening with basictex could just as easily happen with anything else installed from brew.

Reply:
Agree, I remember we had to use `brew update` for `gcc` for a long time. But I think we should avoid `brew update` when it is possible due to it increases jobs time.

Reply:
Alright, moved it in https://github.com/microsoft/LightGBM/pull/4161/commits/f93d7aae3f085c0e0c564db305694936773e4e44 since this is blocking all other PRs.

But I disagree with treating it as temporary and only doing this in the R package's tests. I think having every Mac build be 15 seconds longer is worth a reduced risk of problems like #4160 blocking PR merges in LightGBM for multiple days.

Reply:
Hmm, I'm very surprised to see 15sec for `brew update`! I remember it took several minutes...
If this command is really executed within <1min nowadays, I'm totally fine to have `brew update` for **each build**.

Reply:
I'll merge this for now to unblock other PRs, I can test the timings and submit another PR in the future to move this back to `setup.sh`. The update took only a few seconds on my Mac but that's not a fair test of how long it would take in CI.